File: app/src/main/java/com/example/justsomeapplication/di/main/MainActivityModule.java
package com.example.justsomeapplication.di.main;

import com.example.justsomeapplication.di.main.fragment.MainFragmentModule;
import com.example.justsomeapplication.ui.main.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public interface MainActivityModule {

    @MainActivityScope
    @ContributesAndroidInjector(modules = {MainFragmentModule.class})
    MainActivity activity();
}

File: app/src/main/java/com/example/justsomeapplication/di/main/MainActivityModule.kt
package com.example.justsomeapplication.di.main

import com.example.justsomeapplication.di.main.fragment.UserFragmentModule
import com.example.justsomeapplication.ui.main.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
interface MainActivityModule {
    @MainScope
    @ContributesAndroidInjector(modules = [UserFragmentModule::class])
    fun activity(): MainActivity?
}